perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -1115,6 +1115,17 @@ has shortlisted_user => (
     },
 );
 
+sub set_duplicate_of {
+    my ($self, $other_id) = @_;
+    $self->set_extra_metadata( duplicate_of => $other_id );
+    my $dupe = $self->result_source->schema->resultset("Problem")->find($other_id);
+    my $dupes_duplicates = $dupe->duplicates;
```

Comment:
Rather than hitting the DB here, could you just push `$self->id` onto the existing `duplicates` list in`$dupe->extra`?